cmd/snap/cmd_run_test.go

```diff
@@ -1257,3 +1262,112 @@ esac
+	defer restore()
+	// Rig the random UUID generator to return this value.
+	uuid := "cc98cd01-6a25-46bd-b71b-82069b71b770"
+	restore = snaprun.MockRandomUUID(uuid)
+	defer restore()
+	// Replace interactions with DBus so that only session bus is available and responds with our logic.
+	conn, err := testutil.NewDBusTestConn(func(msg *dbus.Message, n int) ([]*dbus.Message, error) {
+		switch n {
+		case 0:
+			return []*dbus.Message{happyResponseToStartTransientUnit(c, msg, "snap."+uuid+".pkg.app.scope", 312123)}, nil
+		}
+		return nil, fmt.Errorf("unexpected message #%d: %s", n, msg)
+	})
+	c.Assert(err, check.IsNil)
+	restore = snaprun.MockDBusSessionBus(conn)
+	defer restore()
+	// Replace the cgroup analyzer function
+	restore = snaprun.MockCgroupProcessPathInTrackingCgroup(func(pid int) (string, error) {
+		return "/user.slice/user-12345.slice/user@12345.service/snap." + uuid + ".pkg.app.scope", nil
+	})
+	defer restore()
+
+	err = snaprun.CreateTransientScope("snap.pkg.app")
+	c.Check(err, check.IsNil)
+}
+
+func happyResponseToStartTransientUnit(c *check.C, msg *dbus.Message, scopeName string, pid int) *dbus.Message {
+	// XXX: Those types might live in a package somewhere
+	type Property struct {
+		Name  string
+		Value interface{}
+	}
+	type Unit struct {
+		Name  string
+		Props []Property
+	}
+	// Signature of StartTransientUnit, string, string, array of Property and array of Unit (see above).
+	requestSig := dbus.SignatureOf("", "", []Property{}, []Unit{})
+
+	c.Assert(msg.Type, check.Equals, dbus.TypeMethodCall)
+	c.Check(msg.Flags, check.Equals, dbus.Flags(0))
+	c.Check(msg.Headers, check.DeepEquals, map[dbus.HeaderField]dbus.Variant{
+		dbus.FieldDestination: dbus.MakeVariant("org.freedesktop.systemd1"),
+		dbus.FieldPath:        dbus.MakeVariant(dbus.ObjectPath("/org/freedesktop/systemd1")),
+		dbus.FieldInterface:   dbus.MakeVariant("org.freedesktop.systemd1.Manager"),
+		dbus.FieldMember:      dbus.MakeVariant("StartTransientUnit"),
+		dbus.FieldSignature:   dbus.MakeVariant(requestSig),
+	})
+	c.Check(msg.Body, check.DeepEquals, []interface{}{
+		scopeName,
+		"fail",
+		[][]interface{}{
+			{"PIDs", dbus.MakeVariant([]uint32{uint32(pid)})},
+		},
+		[][]interface{}{},
+	})
+
+	responseSig := dbus.SignatureOf(dbus.ObjectPath(""))
+	return &dbus.Message{
+		Type: dbus.TypeMethodReply,
+		Headers: map[dbus.HeaderField]dbus.Variant{
+			dbus.FieldReplySerial: dbus.MakeVariant(msg.Serial()),
+			dbus.FieldSender:      dbus.MakeVariant(":1"), // This does not matter.
+			// dbus.FieldDestination is provided automatically by DBus test helper.
+			dbus.FieldSignature: dbus.MakeVariant(responseSig),
+		},
+		// The object path returned in the body is not used by snap run yet.
+		Body: []interface{}{dbus.ObjectPath("/org/freedesktop/systemd1/job/1462")},
+	}
+}
+
+func (s *RunSuite) TestDoCreateTransientScopeHappy(c *check.C) {
+	conn, err := testutil.NewDBusTestConn(func(msg *dbus.Message, n int) ([]*dbus.Message, error) {
```

Comment:
(a bit of a drive-by) a common pattern we use in the http tests is that we add `n := 0` in the outside and then `n++` inside the dbus test conn handler and at the end check that the handler was called the expected amount of times. Checking on the outside is important IMO otherwise snaprun.DoCreateTransitionScope() { return nil } would also pass this test.

Reply:
I'm aware of this pattern. The reason I didn't use it here is that the test fixture also participates in `n` counting, handling responses to the `Hello` method that is implicitly called.

An empty handler would _not_ pass, notably, as lack of a response is an error.